src/network/networkd-sysctl.c

```
@@ -40,10 +54,91 @@ static void manager_set_ip_forwarding(Manager *manager, int family) {
+                                *p = '.';
+
+                log_warning("sysctl %s was changed from '%s' to '%s'", path, value, buf);
+        }
+}
+
+static int sysctl_inotify_manager(sd_event_source *source, const struct inotify_event *event, void *userdata) {
+        Manager *manager = ASSERT_PTR(userdata);
+        const char *dirname;
+        int i = 0;
+
+        FOREACH_ELEMENT(src, manager->sysctl_event_sources) {
+                if (source == *src) {
+                        dirname = global_sysctls[i];
+                        sysctl_overwrite_check(source, event, manager->sysctls, dirname);
+                        break;
+                }
+                i++;
+        }
+
+        return 0;
+}
+
+static int sysctl_inotify_link(sd_event_source *source, const struct inotify_event *event, void *userdata) {
+        Link *link = ASSERT_PTR(userdata);
+        const char *dirname;
+        int i = 0;
+
+        FOREACH_ELEMENT(src, link->sysctl_event_sources) {
+                if (source == *src) {
+                        dirname = link_sysctls[i];
+                        sysctl_overwrite_check(source, event, link->sysctls, path_join(dirname, link->ifname));
+                        break;
+                }
+                i++;
+        }
+
+        return 0;
+}
+
+static int manager_inotify_add(Manager *manager) {
+        sd_event_source **source;
+
+        assert(manager);
+
+        source = manager->sysctl_event_sources;
+
+        FOREACH_ELEMENT(p, global_sysctls) {
+                if (sd_event_add_inotify(manager->event, source, *p, IN_CLOSE_WRITE | IN_ONLYDIR, sysctl_inotify_manager, manager) < 0)
+                        return log_error_errno(errno, "Failed to watch sysctl %s: %m", *p);
+                source++;
+        }
+
+        return 0;
+}
+
 void manager_set_sysctl(Manager *manager) {
         assert(manager);
         assert(!manager->test_mode);
 
+        manager_inotify_add(manager);
```

Missing error handling.

---

it's in `manager_inotify_add()` itself, is it not enough to print a warning?
It's not an error so critical to stop doing other things.

---

it's ok to ignore errors here, but please make that explicit, i.e. cast return param to `(void)`